test/functional/helpers/session.js

```diff
@@ -14,13 +36,22 @@ async function initDriver (caps, adbPort) {
     } catch (ign) {}
   }
 
+  if (CLOUD) {
+    // on cloud tests, we want to set the `name` capability
+    if (!caps.name) {
+      caps.name = process.env.SAUCE_JOB_NAME || process.env.TRAVIS_JOB_NUMBER || 'unnamed';
+    }
+  }
+
   // Create a WD driver
-  logger.debug(`Starting session on ${DEFAULT_HOST}:${DEFAULT_PORT}`);
-  let driver = await wd.promiseChainRemote(DEFAULT_HOST, DEFAULT_PORT);
+  const host = getHost();
+  const port = getPort();
+  logger.debug(`Starting session on ${host}:${port}`);
+  driver = await wd.promiseChainRemote(host, port);
   await driver.init(caps);
 
   // In Travis, there is sometimes a popup
-  if (process.env.CI) {
+  if (CI && !CLOUD) {
```

Comment:
is CI and TRAVIS the same thing? I don't really get the logic behind these conditions

Reply:
Travis sets a number of variables by default. https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

In this case, Travis emulators often get a popup that blocks automation. But when using Sauce, Travis is just running the test, not the emulator. So we need to check for it only on the condition where the emulator is running within Travis' infrastructure.